### ISSUE TITLE
Lat/long ordering in geojson

### ISSUE THREAD

**Issue by psychemedia** [python-visualization/folium#1082]
Lat/long ordering is inconsistent across various packages and tooling when working with points.

It would be useful if `folium.GeoJson()` included a flag `flip=Boolean` that could be set to flip lat and long co-ordinates in a passed geojson file.

**Comment by ocefpaf** [comment]
We have a `_locations_mirror(x)` in `folium.utilities` that does that for any iterable. Maybe we could make that a first class citizen but I'm not sure about overloading the `folium.GeoJson()` with a `kw`.

**Comment by Conengmo** [comment]
yes I’m also hesitant about adding more flags. How about a separate function you can feed your geojson data and that returns the same data but with the coordinates flipped? I reckon something like that already exists in other libraries, maybe you know of anyone?

**Comment by psychemedia** [comment]
My particular use case (at the moment) is a `geopandas` dataframe with a geometry containing `Point` items that are incorrectly ordered for `folium` purposes.

I'd half expected to find a `geopandas` or `shapely` utility to cope with it but I'm not v. familiar with geo packages so there may be something in there somewhere too.

**Comment by mcooper44** [comment]
If it is a geopands dataframe could you not gdf.apply() a  lambda function to swap the order of the points?  so if the coordinates points are in the column 'Coordinates' and you want to swap the x y points gdf['New_Coordinates'] = gdf['Coordinates'].apply(lambda x: Point(x.y, x.x))

**Comment by psychemedia** [comment]
@mcooper44 Ah, thanks for that... but the issue is then I have to keep messing around with the dataframe. I should probably check on the original rationale for the point order; I think it was to match with a dodgy KML import...

**Comment by mcooper44** [comment]
@psychemedia is it a large set of coordinates?  Having a swapped column for mapping and the original column for matching the dodgy KML import may work?

**Comment by eabase** [comment]
This is really weird, why is `folium.GeoJson()` using `(x,y)`, whereas `Polygon()` or `folium.vector_layers.Polygon(locations=...)` is using `(y,x)`? 
What am I missing to understand here?

**Comment by Conengmo** [comment]
Leaflet, which folium is a wrapper for, uses [lat, lon], wheres the geojson standard uses [lon, lat].

https://leafletjs.com/reference.html#latlng

https://en.wikipedia.org/wiki/GeoJSON#Geometries

**Comment by Conengmo** [comment]
I don't think this is something we want to include in folium. Leaflet and GeoJSON have differing standards, folium works with both of them. If you extract coordinates from a GeoJSON for some other purpose, we expect the user to deal with the difference.

**Comment by eabase** [comment]
@Conengmo 
It may be beneficial to the documentation if you'd put a big red ❗  note in the readme about exactly this:
> Leaflet, which folium is a wrapper for, uses [lat, lon], wheres the geojson standard uses [lon, lat].

**Comment by ocefpaf** [comment]
> It may be beneficial to the documentation if you'd put a big red  note in the readme about exactly this:

Do you want to send a PR for it? We could use a note in the docs too. Here is a SO question [for inspiration](https://stackoverflow.com/questions/18636564/lat-long-or-long-lat). And the mandatory West Wing [episode on maps](https://youtu.be/vVX-PrBRtTY?t=189) coordinates and order.

**Comment by aghoshpro** [comment]
https://macwright.com/lonlat/

**Comment by hansthen** [comment]
Reopened as a documentation issue.

**Comment by Seascavenger** [comment]
Hi @hansthen and @Conengmo, I'd like to work on issue #1082. I'm planning to add documentation about the coordinate ordering difference between Leaflet and GeoJSON with conversion examples. Would this be helpful?

**Comment by hansthen** [comment]
Yes, this would be really appreciated.
